✨ Publish New Docs ✨ (#4904)
software-mansion/react-native-reanimated: docs/docs/animations/withTiming.mdx

<summary>Type definitions</summary>

```typescript
type AnimatableValue = number | string | number[];

interface WithTimingConfig {
  duration?: number;
  easing?: EasingFunction;
}

function withTiming<T extends AnimatableValue>(
  toValue: T,
  config?: WithTimingConfig,
  callback?: (finished?: boolean, current?: AnimatableValue) => void
): T;
```

</details>

### Arguments

#### `toValue`

The value on which the animation will come at rest.

#### `config` <Optional />

The timing animation configuration.

import { useTimingPlayground } from '@site/src/components/InteractivePlayground';

<InteractivePlayground usePlayground={useTimingPlayground} />

Available properties:

| Name                  | Type     | Default                     | Description                                           |
| --------------------- | -------- | --------------------------- | ----------------------------------------------------- |
| duration <Optional /> | `number` | 300                         | Length of the animation (in milliseconds).            |
| easing <Optional />   | `Easing` | `Easing.inOut(Easing.quad)` | An easing function which defines the animation curve. |

##### `Easing`

> The `easing` parameter lets you fine-tine the animation over the specified time duration. For example, you can make the animation begin with fast acceleration and then slow it down towards the end, or start slowly then pick up some speed before slowing down again towards the end.

Reviewer: ```suggestion
The `easing` parameter lets you fine-tune the animation over the specified time duration. For example, you can make the animation begin with fast acceleration and then slow down towards the end, or start slowly, then pick up speed before slowing down again towards the end.
```

> It all will start to make sense when you compare side by side a `linear` easing with the default `Easing.inOut(Easing.quad)` easing.

Reviewer: ```suggestion
It will all start to make sense when you compare a `linear` easing side by side with the default `Easing.inOut(Easing.quad)` easing.
```

import TimingEasingCompare from '@site/src/examples/TimingEasingCompare';
import TimingEasingCompareSrc from '!!raw-loader!@site/src/examples/TimingEasingCompare';

<InteractiveExample
  src={TimingEasingCompareSrc}
  component={<TimingEasingCompare />}
/>

Reanimated provides a handful of ready-to-use easing functions in the `Easing` module. You can find a visualization of some common easing functions at [http://easings.net/](http://easings.net/).

> You use our built-in easings by passing them as `easing` property to `withTiming` config:

Reviewer: ```suggestion
You can use our built-in easings by passing them as the `easing` property to the `withTiming` config:
```
